File: app/src/main/java/com/example/myapplication/kt/presenter/TianQiPresenter.java
package com.example.myapplication.kt.presenter;

import com.example.myapplication.base.BaseObserver;
import com.example.myapplication.base.BasePresenter;
import com.example.myapplication.bean.WeatherBean;

public class TianQiPresenter extends BasePresenter<TianQiContract.TianQiView> implements TianQiContract.TianQiPresenter {

    public TianQiPresenter(TianQiContract.TianQiView baseView) {
        super(baseView);
    }

    @Override
    public void getWeatherList(String key, String city) {
          addDisposable(apiServer.getWeatherGet(key, city), new BaseObserver<WeatherBean>(baseView) {

              @Override
              public void onSuccess(WeatherBean o) {
                    baseView.showTianQiList(o);
              }

              @Override
              public void onError(String code, String msg) {

              }
          });
    }
}

File: app/src/main/java/com/example/myapplication/kt/presenter/TianQiPresenter.kt
package com.example.myapplication.kt.presenter

import android.util.Log
import com.example.myapplication.base.BaseObserver
import com.example.myapplication.base.BasePresenter
import com.example.myapplication.bean.WeatherBean
import com.example.myapplication.kt.presenter.TianQiContract.TianQiView

class TianQiPresenter(baseView: TianQiView?) : BasePresenter<TianQiView?>(baseView), TianQiContract.TianQiPresenter {
    override fun getWeatherList(key: String?, city: String?) {
        addDisposable(apiServer.getWeatherGet(key, city), object : BaseObserver<WeatherBean?>(baseView) {
            override fun onError(code: String, msg: String) {
                Log.e("请求",msg+"")
                baseView!!.showMsg(msg)
            }
            override fun onSuccess(o: WeatherBean?) {
                Log.e("请求","success")
                baseView!!.showTianQiList(o)
            }
        })
    }
}